File: programmers/완주하지못한선수.js
function solution(participant, completion) {
    let answer = '';
    let completion_map = {}

    for (let i = 0; i < completion.length; i++) {
        let key =  completion[i]
        if (key in completion_map) {
            completion_map[key] += 1
        }
        else {
            completion_map[key] = 1
        }
    }

    for (let i = 0; i < participant.length; i++) {
        let key = participant[i]
        if ( !(key  in completion_map)) {
            console.log(key)
            return key
        }
        else {
            completion_map[key] -= 1

            if (completion_map[key] == 0) {
                delete completion_map[key]
            }
        }
    }
    
    return answer;
}

solution(["leo", "kiki", 'eden'],['eden', 'kiki'])

File: programmers/완주하지 못한 선수.py
'''
완주하지 못한 선수
https://programmers.co.kr/learn/courses/30/lessons/42576?language=python3
'''
def solution(participant, completion):
    comp_dict = dict()
    for player in completion:
        if player not in comp_dict:
            comp_dict[player] = 1
        else:
            comp_dict[player] += 1
    for player in participant:    
        if player not in comp_dict:
            return player
        else:
            comp_dict[player] -= 1
            if comp_dict[player] == 0:
                del comp_dict[player]